src/neta/matchedpath.cpp

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "neta/matchedpath.h"
+#include "classes/speciesatom.h"
+#include <algorithm>
+
+// Return path of matched atoms
+const std::vector<const SpeciesAtom *> &NETAMatchedPath::path() const { return path_; }
+
+// Return identified atoms (if any) in path
+const std::map<std::string, std::set<const SpeciesAtom *>> &NETAMatchedPath::identifiers() const { return identifiers_; }
+
+/*
+ * Path Management
+ */
+
+// Append atom to path
+void NETAMatchedPath::append(const SpeciesAtom *i) { path_.push_back(i); }
+
+// Return whether the path contains the specified atom
+bool NETAMatchedPath::contains(const SpeciesAtom *i) const { return std::find(path_.begin(), path_.end(), i) != path_.end(); }
+
+// Return whether the specified atom is the root atom in the path
+bool NETAMatchedPath::isRoot(const SpeciesAtom *i) const { return !path_.empty() && (path_.front() == i); }
+
+// Merge supplied path into this one, appending new atoms and copying any relevant identifiers
+void NETAMatchedPath::merge(const NETAMatchedPath &other)
+{
+    // Copy new, unique atoms to our path
+    std::copy_if(other.path_.begin(), other.path_.end(), std::back_inserter(path_),
+                 [&](const auto *j) { return std::find(path_.begin(), path_.end(), j) == path_.end(); });
```

Comment:
```suggestion
    std::copy(other.path_.begin(), other.path_.end(), std::inserter(path_, path_.begin()));
```

Changing `path_` from `std::vector` to `std::set` would clean up the code here and on the `contains` function.  We don't seem to depend on the order of the elements and I can't find any places outside the class implementation where changes would be necessary, but I might be missing something.

Reply:
I have now made this change - we do make one "positional" distinction for atoms, and that is the so-called root atom (where the matching process started off from). This is dealt with by a specific member variable, which actually makes more sense anyway.  At the same time I have renamed the class and some of its functions to reflect better what the class contains (i.e. it is no longer any kind of path).